python/plugins/processing/tools/vector.py

```diff
@@ -485,7 +494,37 @@ def _runSQL(sql):
                                     table=uri.table().lower(), schema=uri.schema(), srid=crs.authid().split(":")[-1],
                                     typmod=GEOM_TYPE_MAP[geometryType].upper()))
 
-            self.layer = QgsVectorLayer(uri.uri(), uri.table(), "postgres")
```

Comment:
Why have you changed this line? this actually breaks the postgis export. The provider should be "postgres".